async-nats/src/client.rs

```diff
@@ -574,3 +579,48 @@ impl Request {
         self
     }
 }
+
+#[derive(Error, Debug)]
+#[error("failed to send subscribe")]
+pub struct SubscribeError(#[from] mpsc::error::SendError<Command>);
+
+#[derive(Debug, Error)]
+pub enum RequestError {
+    #[error("request timed out")]
+    TimedOut,
+    #[error("no responders")]
+    NoResponders,
+    #[error("request error: {0:?}")]
+    Io(IoErrorKind),
+}
+
+#[derive(Debug, Error)]
+pub enum FlushError {
+    #[error("failed to send flush request")]
+    SendError(Box<mpsc::error::SendError<Command>>),
+    #[error("flush failed")]
+    FlushError,
+}
+
+#[derive(Debug, Error)]
+pub enum IoErrorKind {
```

Comment:
Not sure about the cases in this one just yet

Reply:
Neither I'am.
Looking for better alternative over the weekend.